File: 2904-shortest-and-lexicographically-smallest-beautiful-string/2904-shortest-and-lexicographically-smallest-beautiful-string.java
class Solution {
    public String shortestBeautifulSubstring(String s, int k) {
        ArrayList<String> list = new ArrayList<>();
        for (int i = 0; i < s.length(); i++) {
            int ones = 0;
            for (int j = i; j < s.length(); j++) {
                if (s.charAt(j) == '1') {
                    ones++;
                }
                if (ones == k) {
                    list.add(s.substring(i, j + 1));
                }
                if (ones > k) {
                    break;
                }
            }
        }
        if (list.isEmpty()) {
            return "";
        }
        Collections.sort(list, (a, b) -> {
            if (a.length() != b.length()) {
                return Integer.compare(a.length(), b.length());
            }
            return a.compareTo(b);
        });
        return list.get(0);
    }
}

File: 2904-shortest-and-lexicographically-smallest-beautiful-string/2904-shortest-and-lexicographically-smallest-beautiful-string.cpp
class Solution {
public:
    string shortestBeautifulSubstring(string s, int k) {
        vector<string> check;

        for (int i = 0; i < s.length(); i++) {
            int ones = 0;

            for (int j = i; j < s.length(); j++) {

                if (s[j] == '1') {
                    ones++;
                }

                if (ones == k) {
                    check.push_back(s.substr(i, j - i + 1));
                }

                if (ones > k) {
                    break;
                }
            }
        }

        if (check.empty()) {
            return "";
        }

        sort(check.begin(), check.end(), [](string a, string b) {
            if (a.length() != b.length()) {
                return a.length() < b.length();
            }

            return a < b;
        });

        return check[0];
    }
};